spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -47,6 +49,11 @@ class CassandraJoinRDD[L, R] private[connector](
 
   override protected val classTag = rightClassTag
 
+  override protected[connector] lazy val rowReader: RowReader[R] = manualRowReader match {
+    case Some(rowReader) => rowReader
```

Comment:
Agreed, but the other solution will be much more disruptive (do we have time for 1.4?)